### PR TITLE
[Tests] Onboarding Integration tests 

### PR DESCRIPTION
**Pull request by mennenia**
Having a go at #2053.

FYI, when recording a UI test, the `setUp` method won't be run, so it doesn't necessarily start in the "blank slate". You may want to manually reset the contents and settings of the device, depending on where you want the app to start (if not in onboarding, then it's probably fine :)).

### CODE REVIEW

**Comment by mennenia** [comment]
🍏 Greeeeeennnn on CI!!! 🎉 💃 🍾

**Comment by alloy** [comment]
# Happy Tester Panda!

Giphy

File: Artsy/App/ARAppDelegate.m [hunk 1]
```diff
@@ -86,7 +86,9 @@ + (ARAppDelegate *)sharedInstance
 
 - (BOOL)application:(UIApplication *)application willFinishLaunchingWithOptions:(NSDictionary *)launchOptions
 {
-    if (ARIsRunningInDemoMode) {
+    if (ARIsRunningInDemoMode ||
+        [[[NSProcessInfo processInfo] environment][@"TEST_SCENARIO"] isEqual:@"ONBOARDING"]) {
```
Comment (alloy) [inline Artsy/App/ARAppDelegate.m]: This doesn’t need to check for a specific scenario, if it’s not `nil` then we’re testing.

**Comment by alloy** [comment]
I'm thinking that we should do the user manager stuff all in the scenario file.  So if there's a test scenario environment variable, just run the scenario and then if it's an on boarding one ensure the state is signed out and otherwise signed in. That keeps it all nicely contained in the one file.

File: ArtsyRecordedTests/ArtsyRecordedTests.m [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    [emailTextField typeText:randomEmail];
+    [nextButton2 tap];
+
+    XCUIElement *completeButton = app.buttons[@"COMPLETE"];
+
+    // We haven't filled out the password yet, so we cannot continue
+    XCTAssert(!completeButton.isEnabled);
+
+    XCUIElement *passwordSecureTextField = app.secureTextFields[@"Password"];
+    [passwordSecureTextField typeText:@"sosecurewoo"];
```
Comment (orta) [inline ArtsyRecordedTests/ArtsyRecordedTests.m]: hah

File: Artsy/App/ARAppDelegate.m [hunk 3]
```diff
@@ -87,9 +88,14 @@ + (ARAppDelegate *)sharedInstance
 - (BOOL)application:(UIApplication *)application willFinishLaunchingWithOptions:(NSDictionary *)launchOptions
 {
     if (ARIsRunningInDemoMode) {
+        [[ARUserManager sharedManager] disableSharedWebCredentials];
```
Comment (mennenia) [inline Artsy/App/ARAppDelegate.m]: Decided to keep this, because in demo mode we don't want to display web creds either I think.

**Comment by mennenia** [comment]
@alloy sorry this is so late >_< but split off the test scenarios into another file :) Good to be reviewed from my end!

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          2 Errors
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">No CHANGELOG changes made</td>
    </tr>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Developer Specific file shouldn’t be changed</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Needs testing on a Phone if change is non-trivial</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Comment by mennenia** [comment]
Tests are having a real field day :S. Went all green this morning with my main commits, then I added changelog, and one of the UI tests failed (passes locally). I'm rebuilding etc, but still trying to figure this out.

**Comment by orta** [comment]
One idea we talked about trying was to separate the build commands for UI testing + unit testing, so that we can give a timeout period between test

**Comment by alloy** [comment]
Ok, waiting to further review until I hear we’re good on CI 😞

**Comment by mennenia** [comment]
Unsure what to do with this... Left a more [extensive comment here](https://github.com/artsy/eigen/issues/2053#issuecomment-273206439).

**Comment by alloy** [comment]
As discussed on #2053, can you remove them from being run on CI, for now?

**Comment by mennenia** [comment]
Ok @alloy I think this is good - I'm not sure what Developer Specific file Danger is erroring about though ._.

File: circle.yml [hunk 4]
```diff
@@ -1,4 +1,4 @@
-machine:
+achine:
```
Comment (alloy) [inline circle.yml]: Missing m

**Comment by mennenia** [comment]
@alloy yeah that's already fixed :) The danger check is about something else

**Comment by orta** [comment]
It's the scheme: `Artsy.xcodeproj/xcshareddata/xcschemes/Artsy.xcscheme` 

This is fine to ignore 👍

**Comment by mennenia** [comment]
Ah cool, cheers :)

**Comment by alloy** [comment]
Ah yeah, was looking at old diff…

Ace, merging!

**Pull request merged**